### PR TITLE
[#8] [Backend] As a user, I can login - PART 2 - Mocks

### PR DESCRIPTION
**Pull request by ducbm051291**
#8 

## What happened

The user should be able to login by calling Login API.
 
## Insight

- Create Resolver+Data extension to register services, use case for injection.
- Create SessionRepository
- Create StoreTokenUseCase
- Create Resolver+Domain extension and Resolver+ResolverRegistering to register injection services, domains, and data
- Create extension for publisher to convert to Driver and Observable
- Update typealias Driver
- Create KeychainProtocolMock
- Create APIToken+Dummy
- Create Resolver extension in Test to register injections
 
## Proof Of Work

- N/A

### CODE REVIEW

File: Surveys/Sources/Data/Repositories/AuthenticationRepository.swift [hunk 1]
```diff
@@ -12,7 +12,7 @@ import Resolver
 
 final class AuthenticationRepository: AuthenticationRepositoryProtocol {
 
-    @Injected private var networkAPI: NetworkAPIProtocol
+    @Injected(name: .defaultNetworkAPI) private var networkAPI: NetworkAPIProtocol
```
Comment (blyscuit) [inline Surveys/Sources/Data/Repositories/AuthenticationRepository.swift:15]: Do we need to add the name, I only see one kind of networkAPI. Also is the name declared for the main app in another PR?

File: Surveys/Sources/Supports/Extensions/Combine/Publisher+Driver.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    func empty() -> Driver<Output> {
+        Empty().eraseToAnyPublisher()
+    }
+
+    func asDriver() -> Driver<Output> {
+        self.catch { _ in Empty() }
+            .receive(on: RunLoop.main)
+            .eraseToAnyPublisher()
+    }
```
Comment (blyscuit) [inline Surveys/Sources/Supports/Extensions/Combine/Publisher+Driver.swift:26]: Don't see some of these being call here 🤔

File: Surveys/Sources/Supports/Helpers/Typealiases/Typealiases.swift [hunk 3]
```diff
@@ -13,3 +13,4 @@ typealias CancelBag = Set<AnyCancellable>
 typealias HTTPHeaders = [String: String]
 typealias Localize = R.string.localizable
 typealias Observable<T> = AnyPublisher<T, Error>
+typealias Driver<T> = AnyPublisher<T, Never>
```
Comment (blyscuit) [inline Surveys/Sources/Supports/Helpers/Typealiases/Typealiases.swift:16]: I suggest adding comment for Observable and Driver for quick documentation.
```suggestion
/// Publisher with error `AnyPublisher<T, Error>`
typealias Observable<T> = AnyPublisher<T, Error>
/// Publisher with no error `AnyPublisher<T, Never>`
typealias Driver<T> = AnyPublisher<T, Never>
```

File: SurveysTests/Sources/Injection/Resolver+Domain.swift [hunk 4]
```diff
@@ -0,0 +1,9 @@
+//
+//  Resolver+Domain.swift
+//  SurveysTests
+//
+//  Created by David Bui on 17/11/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Foundation
```
Comment (blyscuit) [inline SurveysTests/Sources/Injection/Resolver+Domain.swift:9]: This file is empty again 🤔

File: Surveys/Sources/Data/Repositories/AuthenticationRepository.swift (hunk 1 above)
Comment (ducbm051291) [reply Surveys/Sources/Data/Repositories/AuthenticationRepository.swift:15]: @blyscuit hehe, sorry. I copied a lot from Worpt project. In Worpt, we have 3 network ap

File: Surveys/Sources/Supports/Extensions/Combine/Publisher+Driver.swift (hunk 2 above)
Comment (ducbm051291) [reply Surveys/Sources/Supports/Extensions/Combine/Publisher+Driver.swift:26]: It's saved for future when we integrate view model

File: Surveys/Sources/Supports/Helpers/Typealiases/Typealiases.swift (hunk 3 above)
Comment (ducbm051291) [reply Surveys/Sources/Supports/Helpers/Typealiases/Typealiases.swift:16]: Good suggestion.

**Review by blyscuit — APPROVED** [review]
Looks good.

**Review by Shayokh144 — APPROVED** [review]

**Pull request merged**